string operation
string door_tag
string door_dir
int16 door_index
-----------------
bool return_val


